diff --git a/src/util/os_memory_aligned.h b/src/util/os_memory_aligned.h
index 99a0b20..1f1dcb5 100644
--- a/src/util/os_memory_aligned.h
+++ b/src/util/os_memory_aligned.h
@@ -35,6 +35,7 @@
 #error "Must not be included directly. Include os_memory.h instead"
 #endif
 
+#include "detect_os.h"
 
 /**
  * Add two size_t values with integer overflow check.
@@ -48,7 +49,7 @@ add_overflow_size_t(size_t a, size_t b, size_t *res)
 }
 
 
-#if defined(HAVE_POSIX_MEMALIGN)
+#if defined(HAVE_POSIX_MEMALIGN) && !DETECT_OS_STANIX
 
 static inline void *
 os_malloc_aligned(size_t size, size_t alignment)
